File: idling-resource-dialog-fragment/app/src/main/java/com/sqisland/espresso/idling_resource/dialog_fragment/MainActivity.java
package com.sqisland.espresso.idling_resource.dialog_fragment;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {
  private TextView textView;

  @Override
  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.activity_main);

    textView = findViewById(R.id.text);

    LoadingDialogFragment fragment = new LoadingDialogFragment();
    fragment.setCancelable(false);
    fragment.show(getSupportFragmentManager(), LoadingDialogFragment.TAG);
  }

  public void onLoadingFinished() {
    textView.setText(R.string.done);
  }
}

File: idling-resource-dialog-fragment/app/src/main/java/com/sqisland/espresso/idling_resource/dialog_fragment/MainActivity.kt
package com.sqisland.espresso.idling_resource.dialog_fragment

import android.os.Bundle
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {
  private lateinit var textView: TextView

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    setContentView(R.layout.activity_main)

    textView = findViewById(R.id.text)

    val fragment = LoadingDialogFragment()
    fragment.isCancelable = false
    fragment.show(supportFragmentManager, LoadingDialogFragment.TAG)
  }

  fun onLoadingFinished() {
    textView.setText(R.string.done)
  }
}